File: python/fibonacci.py
#!/usr/bin/python

import sys

# Iterate and print example
def fib_print(n):
    # Set seed values
    f_prev = 0
    f_curr = 1
    # Print seed values
    sys.stdout.write("%d %d" % (f_prev, f_curr))
    # Print the remaining n-2 values
    i = 0
    while i < n - 2:
        f_curr, f_prev = f_curr + f_prev, f_curr
        sys.stdout.write(" %d" % f_curr)
        i += 1
    print

# List example
def fib_list(n):
    if n < 2:
        return []
    # Set seed values
    f = [0, 1]
    # Collect the remaining n-2 values
    i = 0
    while i < n - 2:
        i += 1
        f.append(f[len(f)-1] + f[len(f)-2])
    # Return list
    return f

fib_print(20)
print fib_list(20)

File: ruby/lib/fibonacci.rb
# Fibonacci Class
class Fibonacci

  def initialize(n)
    @n = n
  end

  # Fibonacci String
  def fib_print
    # Set seed values
    f_prev = 0
    f_curr = 1
    # Print first value
    f = format '%d', f_prev
    # Print 19 more values
    i = 0
    while i < @n - 1
      i += 1
      f += format ' %d', f_curr
      f_curr, f_prev = f_curr + f_prev, f_curr
    end
    f
  end

  # Fibonacci Array
  def fib_array
    return [] if @n < 2
    # Set seed values
    f = [0, 1]
    # Collect 18 more values
    Array.new(@n - f.length) { f[f.length] = f[f.length - 1] + f[f.length - 2] }
    # Return array
    f
  end

end